base/applications/cleanmgr/dialog.c

```diff
@@ -0,0 +1,301 @@
+                    break;
+            }
+            break;
+
+        case WM_CLOSE:
+            EndDialog(hwnd, IDCANCEL);
+            break;
+
+        default:
+        {
+            if (message == CleanmgrWindowMsg)
+            {
+                SetForegroundWindow(hwnd);
+            }
+            return FALSE;
+        }
+    }
+    return TRUE;
+}
+
+INT_PTR CALLBACK TabParentDlgProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    WCHAR TempText[ARR_MAX_SIZE] = { 0 };
+
+    switch (message)
+    {
+        case WM_INITDIALOG:
+        {
+            HICON hbmIcon = LoadIconW(GetModuleHandleW(NULL), MAKEINTRESOURCE(IDI_CLEANMGR));
+            if (!lParam)
+            {
+                WCHAR FullText[ARR_MAX_SIZE] = { 0 };
+                LoadStringW(GetModuleHandleW(NULL), IDS_CHOICE_DLG_TITLE, TempText, _countof(TempText));
+                StringCchPrintfW(FullText, sizeof(FullText), TempText, DriveLetter);
+                SetWindowTextW(hwnd, FullText);
+            }
+
+            InitTabControl(hwnd, lParam);
+            SendMessageW(hwnd, WM_SETICON, ICON_SMALL, (LPARAM)hbmIcon);
+            SendMessageW(hwnd, WM_SETICON, ICON_BIG, (LPARAM)hbmIcon);
+            return TRUE;
+        }
+
+        case WM_NOTIFY:
+        {
+            LPNMHDR pnmh = (LPNMHDR)lParam;
+            if ((pnmh->hwndFrom == DialogHandle.hTab) &&
+                (pnmh->idFrom == IDC_TAB) &&
+                (pnmh->code == TCN_SELCHANGE))
+            {
+                TabControlSelChange();
+                break;
+            }
+            return ThemeHandler(hwnd, (LPNMCUSTOMDRAW)lParam);
+        }
+
+        case WM_THEMECHANGED:
+            InvalidateRect(hwnd, NULL, FALSE);
+            break;
+
+        case WM_COMMAND:
+            switch (LOWORD(wParam))
+            {
+                case IDOK:
+                {
+                    LoadStringW(GetModuleHandleW(NULL), IDS_CONFIRMATION, TempText, _countof(TempText));
+                    int MesgBox = MessageBoxW(hwnd, TempText, L"Warning", MB_YESNO | MB_ICONWARNING | MB_DEFBUTTON2);
+                    switch (MesgBox)
+                    {
+                        case IDYES:
+                            EndDialog(hwnd, IDOK);
+                            break;
+
+                        case IDNO:
+                            break;
+                    }
+                    break;
+                }
+                case IDCANCEL:
+                    EndDialog(hwnd, IDCANCEL);
+                    break;
+            }
+            break;
+
+        case WM_CLOSE:
+            EndDialog(hwnd, IDCANCEL);
+            break;
+
+        case WM_DESTROY:
+            if (DialogHandle.hChoicePage)
+            {
+                DestroyWindow(DialogHandle.hChoicePage);
+            }
+            else if (DialogHandle.hOptionsPage)
+            {
+                DestroyWindow(DialogHandle.hOptionsPage);
+            }
+            break;
```

Comment:
Not clear: Is it possible to have both hChoicePage and hOptionsPage window handles being valid?